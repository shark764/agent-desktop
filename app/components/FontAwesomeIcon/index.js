/**
*
* FontAwesomeIcon
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';

import 'font-awesome/css/font-awesome.css';

function FontAwesomeIcon(props) {
  return (
    <i
      id={props.id ? props.id : `${props.name}-icon`}
      className={`fa fa-${props.name} fa-${props.faSize ? props.faSize : '2'}x`}
      style={props.style}
      onClick={props.onclick}
      alt={props.alt || props.name}
    ></i>
  );
}

FontAwesomeIcon.propTypes = {
  name: PropTypes.string.isRequired,
  faSize: PropTypes.string,
  alt: PropTypes.string,
  style: PropTypes.object,
  id: PropTypes.string.isRequired,
  onclick: PropTypes.func,
};

export default Radium(FontAwesomeIcon);
