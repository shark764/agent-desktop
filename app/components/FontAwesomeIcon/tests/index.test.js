/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import FontAwesomeIcon from '../index';

describe('<FontAwesomeIcon />', () => {
  it('should render correctly', () => {
    const rendered = shallow(<FontAwesomeIcon name="globe" id="globeIcon" />);
    expect(rendered).toMatchSnapshot();
  });
  it('should render correctly with additional props', () => {
    const rendered = shallow(
      <FontAwesomeIcon
        name="globe"
        faSize="3"
        alt="testing icon"
        style={{ color: 'blue' }}
        id="globeIcon"
        onclick={() => {
          console.log('test');
        }}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
