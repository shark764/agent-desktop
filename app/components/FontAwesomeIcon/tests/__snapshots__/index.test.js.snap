// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FontAwesomeIcon /> should render correctly 1`] = `
<button
  style={
    Object {
      "backgroundColor": "inherit",
      "border": "none",
      "cursor": "pointer",
      "padding": "0",
    }
  }
  tabIndex={0}
  type="button"
>
  <i
    alt="globe"
    className="fa fa-globe fa-2x"
    id="globeIcon"
  />
</button>
`;

exports[`<FontAwesomeIcon /> should render correctly with additional props 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "inherit",
      "border": "none",
      "cursor": "pointer",
      "padding": "0",
    }
  }
  tabIndex={0}
  type="button"
>
  <i
    alt="testing icon"
    className="fa fa-globe fa-3x"
    id="globeIcon"
    style={
      Object {
        "color": "blue",
      }
    }
  />
</button>
`;
